### DISCUSSION TITLE
Set cellacdc run to use different Qt bindings then PyQt6

### DISCUSSION THREAD

**Discussion by brchn6** [SchmollerLab/Cell_ACDC#505] [Q&A] [answered]
Hello i cant work with PyQt6 n the linux machine in working on, there for I want to set up the application to work on different Qt.

Here the message indicated the options:
```
(acdc) [barc@hgn14 ~]$ cellacdc
------------------------------------------------------------
Cell-ACDC needs to install a GUI library (default library is `PyQt6`).

You can install it now or you can close (press "n") and install
a compatible GUI library with one of the following commands:

    * pip install PyQt6
    * pip install PyQt5 (or `conda install pyqt`)
    * pip install PySide2
    * pip install PySide6

Note: if `PyQt6` installation fails, you could try installing any of the other libraries.

```
Heres the pip list in my acdc env:
```
(acdc) [barc@hgn14 ~]$ pip list |grep Py
PyQt5                  5.15.9
PyQt5-Qt5              5.15.2
PyQt5-sip              12.13.0
PySide2                5.15.2.1
QtPy                   2.4.1
```

**Comment by ElpadoCan** [comment]
Hi @brchn6, I'm sorry to hear you are having issues.

I see that you have both `PyQt5` and `QtPy` so Cell-ACDC should not complain about a missing Qt library. 

The reason why it complains is that some error happens when importing `from qtpy.QtCore import Qt`. Since Cell-ACDC does not display the actual error, we need to diagnose that differently. Can you please activate the `acdc` environment, run the command `python3` (or just `python` if you are using conda), and then run `from qtpy.QtCore import Qt`. This way you should get the actual error in the terminal.

Let me know what the error is, thanks! 

Francesco

**Reply by brchn6** [reply]
Hey, thanks for yor reply, 
here what you asked:
```
(acdc) [barc@hgn22 ~]$ python3
Python 3.9.19 (main, Mar 21 2024, 17:11:28)
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from qtpy.QtCore import Qt
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/labs/mayalab/barc/.conda/envs/acdc/lib/python3.9/site-packages/qtpy/__init__.py", line 287, in <module>
    raise QtBindingsNotFoundError from None
qtpy.QtBindingsNotFoundError: No Qt bindings could be found
>>>
```

**Reply by ElpadoCan** [reply, marked as answer]
Ok so the error appears to be that `qtpy` does not find the Qt bindings (despite the fact they are actually installed). Do you mind testing in a new environment? Please try the following:

0. Update conda with `conda update conda`
1. Create a new env with the command `conda create -n test python=3.10`
2. Activate with `conda activate test`
3. Install PyQt5 with `conda install -c conda-forge pyqt`
4. Install qtpy with `conda install -c conda-forge qtpy`

After this try again running the `python` command and `import qtpy.QtCore`.

If it's imported correctly (no error) you can install cellacdc in this test environment, or if you prefer an environment called `acdc` you need to remove the old `acdc` with the command `conda env remove -n acdc` and recreate it with the same steps above. 

Let me know how it goes thanks!

**Reply by brchn6** [reply]
Hi, Super thanks. 
That was very simple, and it was totally the error I had received before 

the app is up and running 🏃😎
